File: src/org/elixir_lang/structure_view/element/Timed.java
package org.elixir_lang.structure_view.element;

import org.jetbrains.annotations.NotNull;

public interface Timed {
    /**
     * When the defined call is usable
     *
     * @return {@link Time#COMPILE} for compile time ({@code defmacro}, {@code defmacrop});
     *   {@link Time#RUN} for run time {@code def}, {@code defp})
     */
    @NotNull
    Time time();

    enum Time {
        COMPILE,
        RUN
    }
}

File: src/org/elixir_lang/Timed.kt
package org.elixir_lang

import org.elixir_lang.semantic.call.definition.clause.Time

interface Timed {
    /**
     * When the defined call is usable
     *
     * @return [Time.COMPILE] for compile time (`defmacro`, `defmacrop`);
     * [Time.RUN] for run time `def`, `defp`)
     */
    val time: Time
}
